main.tf

```diff
@@ -137,6 +135,7 @@ resource "github_repository" "repository" {
   lifecycle {
     ignore_changes = [
       auto_init,
+      branches,
```

Comment:
Although I understand the intend of introducing this key, it's [not compatible](https://github.com/integrations/terraform-provider-github/pull/1117) with v5. In v5 this logic has been refactored into it's own datasource.

Reply:
Reverted this change. Initially, I added since it was causing a perpetual diff with the v4 provider; however, it is fixed with the latest v5 provider.